src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -636,13 +637,66 @@ HasComplexJoinOrder(Query *queryTree)
 			hasComplexJoinOrder = true;
 			break;
 		}
+
+		/*
+		 * join tree structure is more important for outer joins. right branch
```

Comment:
Does it not matter for inner joins, or does this situation not happen?

(ultra minor: comment should start with capital letter)